File: 01/point.cc
#include <iostream>
using namespace std;

struct Point {
	double x_;
	double y_;
	
	double get_x() {
		return x_;
	}

	double get_y() {
		return y_;
	}
};


int main() {

	Point p1;
	p1.x_ = 1.0;
	p1.y_ = 2.0;

	cout << p1.get_x() << " " << p1.get_y() << endl;
	
	Point p2;
	p2.x_ = 3.0;
	p2.y_ = 6.0;
	cout << p2.get_x() << " " << p2.get_y() << endl;
	
	
	return 0;
}

File: LearnJava01/src/org/elsys/java/Point.java
package org.elsys.java;

public class Point {

	private double x;
	private double y;
	
	public Point(double x, double y) {
		 this.x = x;
		 this.y = y;
	}
	
	public Point() {
		this(0.0, 0.0);
	}
	
	public double getX() {
		return x;
	}
	
	public double getY() {
		return y;
	}
	
	@Override
	public String toString() {
		return "Point(" + x + ", " + y + ")";
		
	}
	
	public static void main(String[] args) {
		Point p0 = null;
		Point p1 = new Point(1.0, 1.0);
		
		System.out.println("p0=" + p0);
		System.out.println("p1=" + p1);
		
		Point p = p1;
		System.out.println("p=" + p);
	}
}
